README.md

```diff
@@ -1,9 +1,176 @@
+        }
+    }
+}
+```
+
+Next, add the extension to the project dependency list:
+
+```groovy
+dependencies {
+    // ...
+    implementation 'de.wuespace.telestion.extension:mavlink:0.4.0'
+    // ...
+}
+```
+
+And synchronize gradle.
+
+Finished!
+
+## Usage
+
+The extension provides different verticles, that you can use and set up in your project `config.json`.
+You will need:
+
+- [Mavlink v2 Validator](https://wuespace.github.io/telestion-extension-mavlink/de/wuespace/telestion/extension/mavlink/ValidatorMavlink2.html)
+- [Mavlink Message Registrar](https://wuespace.github.io/telestion-extension-mavlink/de/wuespace/telestion/extension/mavlink/MavlinkRegistrar.html)
+- [Payload Parser](https://wuespace.github.io/telestion-extension-mavlink/de/wuespace/telestion/extension/mavlink/PayloadParser.html)
+
+The `Mavlink Validator` verticles for `Mavlink v1` and `Mavlink v2` validate the incoming encoded mavlink message
+and sends it to the parser on a valid mavlink message.
+Additionally, all encoded messages are sent to the `Mavlink Logger`
+which stores the encoded or raw version in the preconfigured data file.
+
+A `Mavlink v2 Validator` configuration could look like:
+
+```json
+{
+  "org.telestion.configuration": {
+    "app_name": "Basic Example",
+    "verticles": [
+      {
+        "name": "Mavlink v2 Validator",
+        "verticle": "de.wuespace.telestion.extension.mavlink.ValidatorMavlink2",
+        "magnitude": 1,
+        "config": {
+          "inAddress": "nice-serial",
+          "packetOutAddress": "mavlink-raw-logger",
+          "parserInAddress": "mavlink-payload-parser-in",
+          "password": "AQ=="
+        }
+      }
+    ]
+  }
+}
+```
+
+The mavlink message is received from the `nice-serial` channel, validated
+and logged to the `Mavlink Raw Logger`.
+If the mavlink is valid, the validated message and metadata are sent to the `Payload Parser`.
+
+The configuration of the `Payload Parser` could look like:
+
+```json
+{
+  "org.telestion.configuration": {
+    "app_name": "Basic Example",
+    "verticles": [
+      {
+        "name": "Payload Parser",
+        "verticle": "de.wuespace.telestion.extension.mavlink.PayloadParser",
+        "magnitude": 1,
+        "config": {
+          "inAddress": "mavlink-payload-parser-in",
+          "outAddress": "parsed-payload"
+        }
+      }
+    ]
+  }
+}
+```
+
+This verticle parses the encoded mavlink message incoming from the `mavlink-payload-parser-in` channel
+and the parsed output are sent onto `parsed-payload` channel to other verticles.
+
+It can only parse the encoded message to registered mavlink messages.
+This is done with the `Mavlink Message Registrar` verticle.
+
+This could look like:
+
+```json
+{
+  "org.telestion.configuration": {
+    "app_name": "Basic Example",
+    "verticles": [
+      {
+        "name": "Mavlink Message Registrar",
+        "verticle": "de.wuespace.telestion.project.daedalus2.mavlink.MavlinkRegistrar",
```

Comment:
See here: It was part of the Daedalus2-project. Maybe we want to commit this to this repo as well before finally updating it and writing a proper solution for this problem.

Reply:
Please see comment above :wink:

Reply:
Done in [`a65d67c` (#37)](https://github.com/wuespace/telestion-extension-mavlink/pull/37/commits/a65d67c22f110ff2e2146cc1a87423f83c8d3a98)